sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -1405,6 +1405,31 @@ abstract class DDLSuite extends QueryTest with DDLSuiteBase {
     }
   }
 
+  test("SPARK-51747: When legacy conf enabled data source read ignores option change") {
```

Comment:
to me , it would be clearer to combine the test:
```
 Seq("true", "false").foreach { ignoreCache =>
    withSQLConf(SQLConf.READ_DATA_SOURCE_CACHE_IGNORE_OPTIONS.key -> ignoreCache) {

       if (ignoreCache)
           checkAnswer(...)
       else
           checkAnswer(...)
    }
```
Then the test can just show the two versions of the output.

I see its in many other suite and think its a nice way to reduce the code and side-by-side see the config behavior change.